src/goto-harness/memory_snapshot_harness_generator.cpp

```diff
@@ -157,18 +162,29 @@ void memory_snapshot_harness_generatort::add_init_section(
 
 void memory_snapshot_harness_generatort::add_assignments_to_globals(
   const symbol_tablet &snapshot,
+  goto_modelt &goto_model,
   code_blockt &code) const
 {
+  recursive_initialization_configt recursive_initialization_config;
+  auto recursive_initialization = util_make_unique<recursive_initializationt>(
+    recursive_initialization_config, goto_model);
+
   for(const auto &pair : snapshot)
   {
     const symbolt &symbol = pair.second;
 
     if(!symbol.is_static_lifetime)
       continue;
 
-    code_assignt code_assign(symbol.symbol_expr(), symbol.value);
-
-    code.add(code_assign);
+    if(variables_to_havoc.count(symbol.base_name) == 0)
+    {
+      code_assignt code_assign(symbol.symbol_expr(), symbol.value);
+      code.add(code_assign);
```

Comment:
Nit picking: I don't think the `code_assign` temporary adds value, you might just do `code.add(code_assignt{...});`. If you do prefer to keep it I'd suggest to use `code.add(std::move(code_assign));`

Reply:
Done.